addons/binding/org.openhab.binding.elerotransmitterstick/src/main/java/org/openhab/binding/elerotransmitterstick/handler/EleroChannelHandler.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.elerotransmitterstick.handler;
+
+import static org.openhab.binding.elerotransmitterstick.EleroTransmitterStickBindingConstants.*;
+
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StopMoveType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.elerotransmitterstick.stick.CommandType;
+import org.openhab.binding.elerotransmitterstick.stick.ResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EleroChannelHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Volker Bier - Initial contribution
+ */
+public class EleroChannelHandler extends BaseThingHandler implements StatusListener {
+    private final Logger logger = LoggerFactory.getLogger(EleroChannelHandler.class);
+
+    protected int[] channelIds;
+    protected EleroTransmitterStickHandler bridge;
+
+    public EleroChannelHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        bridge = (EleroTransmitterStickHandler) getBridge().getHandler();
+
+        setChannelIds();
+        for (int channelId : channelIds) {
+            bridge.addStatusListener(channelId, this);
+        }
+
+        if (bridge.getThing().getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
```

Comment:
where do you set the `thing` to `ONLINE`?

Reply:
in line 115

Reply:
In line 115 is is where you first receive a status message, right? If you do not have any information about the thing status before you should set it to `UNKNOWN` at the end of the `initalize` method.